### ISSUE TITLE
Soft wrap not applying on diagnostic messages lines

### ISSUE THREAD

**Issue by mikysett** [zed-industries/zed#18501]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When showing diagnostics on buffers with soft wrap, the diagnostic lines are not wrapped.
I don't know if it's intended and if it can be tweaked in the settings, but in some situations this behavior prevent the user for seeing full error message without toggling the zoom on current page.

### Environment

Zed: v0.154.3 (Zed)
OS: Linux X11 ubuntu 24.04
Memory: 15.4 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics (CML GT2) || Intel open-source Mesa driver || Mesa 24.0.9-0ubuntu0.1

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![image](https://github.com/user-attachments/assets/c71367ab-24c2-48fb-98bc-aa7d7741ba97)

**Comment by maximsnoep** [comment]
I would like to have this fixed too.. Very annoying to have the error and warning messages floating outside of the view. They should be wrapped similar as other content with options like "soft_wrap": "bounded".
